### PR TITLE
Add new quest structs

### PR DESCRIPTION
**Pull request by ddelgado60**
Updated and created new structs and union types for quests for our issue on Sprint 3. Later during Sprint 3, we will try to implement and edit existing functions in quests_state.c to reflect these changes, since the changes here will probably break existing code.

### CODE REVIEW

File: include/quests/quests_structs.h [hunk 1]
```diff
@@ -8,55 +8,145 @@
...
- * This union represents an achievement.
+ * This struct represents an achievement.
+ * 
+ * Components:
+ *  mission: mission to be completed
+ *  completed: bool for if achievement is completed
  */
 typedef struct achievement {
     mission_t *mission;
+    mission_type_t* mission_type;
```
Comment (zsp255) [inline include/quests/quests_structs.h:81]: Where is mission_type_t defined

Comment (zsp255) [reply include/quests/quests_structs.h:81]: nvm

File: include/quests/quests_structs.h [hunk 2]
```diff
@@ -8,55 +8,145 @@
...
+/* 
+ * This struct represents a reward for completing a quest.
+ *
+ * Components:
+ *  xp: an xp amount gained
+ *  item: an item gained
+ */
+typedef struct reward{
+   int xp;
+   item_t *item;
```
Comment (zsp255) [inline include/quests/quests_structs.h:106]: not sure if money is implemented into the game engine. perhaps we could think about adding an additional field next sprint

**Review by zsp255 — APPROVED** [review]
This commit looks quite good. As you said it will probably present errors in compiling and/or failures in our tests. I think it can be merged however, as long as the old version is archived.

File: include/quests/quests_structs.h (hunk 1 above)
Comment (lilyehsani) [inline include/quests/quests_structs.h:81]: Move * over so it's `*mission_type`

**Review by lilyehsani — CHANGES_REQUESTED** [review]
The code and documentation updates look good for the most part, but the compilation errors must be fixed before this can be merged.

File: include/quests/quests_structs.h [hunk 3]
```diff
@@ -8,55 +8,145 @@
...
 
+/*
+ * This union represents a mission.
+ * 
+ * Components:
+ *  a_mission: an active mission
+ *  p_mission: a passive mission
+ *
+ */
+typedef union mission {
```
Comment (zsp255) [inline include/quests/quests_structs.h:67]: This will impact the implementation of the trees. Hold off on integration of this code.

**Review by lilyehsani — APPROVED** [review]
LGTM! Approved

**Comment by ddelgado60** [comment]
It was decided to close this PR and move the added data on this PR to implement_new_quest_fcts branch. This branch branches off of quests/implement-trees since it relies on work done in that branch. We thought it would make more sense to do this, and so this PR is closed.

**Comment by lilyehsani** [comment]
Issue Score: ✔️+

Comments:
Great work overall! I have a few comments:
- Make sure to add anyone who is working on this PR as an Assignee
- Make sure to assign new PRs to the sprint they are worked on (by selecting the appropriate sprint under "Milestone")
- Make sure to select your team's identifier under "Labels"

**Pull request closed without merging**